### PR TITLE
OSDOCS-1977: Adding a 'Preparing to install on RHV' assembly

### PR DESCRIPTION
**Pull request by pneedle-rh**
This applies to master and branch/enterprise-4.8.

This PR relates to https://issues.redhat.com/browse/OSDOCS-1977. The PR adds a "Preparing to install on RHV" assembly that provides high level guidance about the different installation methods available to install OpenShift on RHV.

The preview is [here](https://deploy-preview-33460--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/preparing-to-install-on-rhv.html).

### CODE REVIEW

**Comment by pneedle-rh** [comment]
@openshift/team-documentation for peer review please. Thanks!

File: installing/installing_rhv/preparing-to-install-on-rhv.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+* **xref:../../installing/installing_rhv/installing-rhv-user-infra.adoc#installing-rhv-user-infra[Installing a cluster on RHV with user-provisioned infrastructure]**: You can install {product-title} on {rh-virtualization} infrastructure that you provide.
+
+* **xref:../../installing/installing_rhv/installing-rhv-restricted-network.adoc#installing-rhv-restricted-network[Installing a cluster on RHV in a restricted network]**: You can install {product-title} on {rh-virtualization} in a restricted or disconnected network by creating an internal mirror of the installation release content. You can use this method to install a cluster that does not require an active Internet connection to obtain the software components. You can also use this installation method to ensure that your clusters only use container images that satisfy your organizational controls on external content.
```
Comment (mikemckiernan) [inline installing/installing_rhv/preparing-to-install-on-rhv.adoc:26]: I think internet can be lowercase, ISG p348.

**Review by mikemckiernan — APPROVED** [review]
looks good

File: installing/installing_rhv/preparing-to-install-on-rhv.adoc (hunk 1 above)
Comment (pneedle-rh) [reply installing/installing_rhv/preparing-to-install-on-rhv.adoc]: @mikemckiernan thank you for your review and for highlighting this! I have now resolved this.

File: installing/installing_rhv/preparing-to-install-on-rhv.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+* You reviewed details about the xref:../../architecture/architecture-installation.adoc#architecture-installation[{product-title} installation and update] processes.
+
+* You read the documentation on xref:../../installing/installing-preparing.adoc#installing-preparing[selecting a cluster installation method and preparing it for users].
+
+[id="choosing-an-method-to-install-ocp-on-rhv"]
+== Choosing a method to install {product-title} on RHV
+
+You can install {product-title} on {rh-virtualization-first} by using one of the following deployment methods.
+
+* **xref:../../installing/installing_rhv/installing-rhv-default.adoc#installing-rhv-default[Installing a cluster quickly on RHV]**: You can install {product-title} on {rh-virtualization} infrastructure that is provisioned by the OpenShift Container Platform installation program. You can install a cluster quickly by using the default configuration options.
```
Comment (pneedle-rh) [inline installing/installing_rhv/preparing-to-install-on-rhv.adoc:20]: @emarcusRH can you please confirm if this bullet point is correct from an RHV perspective? Thanks!

Comment (pneedle-rh) [reply installing/installing_rhv/preparing-to-install-on-rhv.adoc]: After reviewing each of the OCP on RHV installation assemblies further, I have updated each of the descriptions to be more explicit.

**Comment by pneedle-rh** [comment]
@sandrobonazzola can you please review this PR from an SME perspective? Thanks!

File: installing/installing_rhv/preparing-to-install-on-rhv.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+* **xref:../../installing/installing_rhv/installing-rhv-user-infra.adoc#installing-rhv-user-infra[Installing a cluster on RHV with user-provisioned infrastructure]**: You can install {product-title} on {rh-virtualization} virtual machines that you provision. You can use the provided Ansible playbooks to assist with the installation.
+
+* **xref:../../installing/installing_rhv/installing-rhv-restricted-network.adoc#installing-rhv-restricted-network[Installing a cluster on RHV in a restricted network]**: You can install {product-title} on {rh-virtualization} in a restricted or disconnected network, by creating an internal mirror of the installation release content. You can use this method to install a user-provisioned cluster that does not require an active internet connection to obtain the software components. You can also use this installation method to ensure that your clusters only use container images that satisfy your organizational controls on external content.
```
Comment (emarcusRH) [inline installing/installing_rhv/preparing-to-install-on-rhv.adoc:26]: I don't think you need the comma in "in a restricted or disconnected network, by creating an internal mirror "

other than that, looks good to me.

Comment (pneedle-rh) [reply installing/installing_rhv/preparing-to-install-on-rhv.adoc:26]: @emarcusRH thank you for your review and for highlighting this. I have now resolved this suggestion.

**Comment by Gal-Zaidman** [comment]
Looks good,
But regarding the Prerequisites section[1] either remove the line:
"You reviewed details about the OpenShift Container Platform installation and update processes"
since we include it in the Prerequisites section of the installation methods

OR

make sure that Prerequisites section is always the same as the installation methods plus the "selecting a cluster installation method and preparing it for users" line

[1] https://deploy-preview-33460--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/preparing-to-install-on-rhv.html#preparing-to-install-on-rhv-prerequisites

**Comment by pneedle-rh** [comment]
> Looks good,
> But regarding the Prerequisites section[1] either remove the line:
> "You reviewed details about the OpenShift Container Platform installation and update processes"
> since we include it in the Prerequisites section of the installation methods
> 
> OR
> 
> make sure that Prerequisites section is always the same as the installation methods plus the "selecting a cluster installation method and preparing it for users" line
> 
> [1] https://deploy-preview-33460--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/preparing-to-install-on-rhv.html#preparing-to-install-on-rhv-prerequisites

@Gal-Zaidman thank you for your review and for this suggestion. My view is that the second option that you suggested is preferable. In the reader journey it is useful to have reviewed the general overview about installations before looking at the "Preparing to install on RHV" page, to determine which installation method for RHV is the most suitable. This is because the page refers to user-provisioned and installer-provisioned installations, which are explained in that page. There is also a chance that readers could be directed straight to an installation assembly without seeing the preparation page. As such, I think it's useful to have the installation overview mentioned as a prerequisite in this assembly and in the individual installation assemblies.

Regarding having the "Selecting a cluster installation method and preparing it for users" prerequisite in all of the installation assemblies, I think that is a great idea. I have subsequently added the prerequisite to each of the OCP on RHV installation assemblies. I will also create a further PR to add it across all of the installation assemblies in the Installation book. Thanks for the suggestion!

**Comment by Gal-Zaidman** [comment]
One thing that conserns me is that a user will see the "Prerequisites" in the "Preparing to install on RHV" and not read the "Prerequisites" in the installation method.
Because "Preparing to install on RHV" can't catch all the installation methods "Prerequisites", maybe we should add a note or call it "General Prerequisites" ? (something like that I hope you got the point :D)

**Comment by pneedle-rh** [comment]
> One thing that conserns me is that a user will see the "Prerequisites" in the "Preparing to install on RHV" and not read the "Prerequisites" in the installation method.
> Because "Preparing to install on RHV" can't catch all the installation methods "Prerequisites", maybe we should add a note or call it "General Prerequisites" ? (something like that I hope you got the point :D)

I saw your most recent message after the merge and cherry pick had been done. The prerequisites section in the "Preparing to install on RHV" assembly are prerequisites that are required prior to having read that assembly specifically. This same format is used throughout the library in many places. There are also "Preparing to install" assemblies now for each of the platform installation subsections. I will discuss your point with our supportability team within the OpenShift docs team and we can then determine whether the current format for prerequisites being specific to a particular assembly needs updating. Thanks again for your feedback.

**Comment by pneedle-rh** [comment]
Running the CP for 4.8 again.

**Comment by pneedle-rh** [comment]
For reference, [here](https://github.com/openshift/openshift-docs/pull/33681) is the PR in which I have added the "Selecting a cluster installation method and preparing it for users" as a prerequisite in all installation assemblies in the Installing book.

**Pull request merged**